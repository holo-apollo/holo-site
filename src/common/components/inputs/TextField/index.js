import React from 'react';
import TextFieldMaterial from '@material-ui/core/TextField';

import FieldWithError from '../FieldWithError';

const TextField = ({ errorText, ...rest }) => (
  <FieldWithError errorText={errorText}>
    <TextFieldMaterial
      {...rest}
      fullWidth={true}
      error={Boolean(errorText)}
      inputProps={{ ...rest.inputProps, maxLength: rest.maxLength }}
      InputLabelProps={{ style: { fontSize: '12px' } }}
      FormHelperTextProps={{ style: { fontSize: '10px' } }}
    />
  </FieldWithError>
);

export default TextField;
