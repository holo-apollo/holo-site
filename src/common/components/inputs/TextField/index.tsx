import React from 'react';
import TextFieldMaterial, { TextFieldProps } from '@material-ui/core/TextField';

import FieldWithError from '../FieldWithError';

type Props = TextFieldProps & {
  errorText?: string;
  maxLength?: number;
};

const TextField = ({ errorText, maxLength, ...rest }: Props) => (
  <FieldWithError errorText={errorText}>
    <TextFieldMaterial
      {...rest}
      fullWidth={true}
      error={Boolean(errorText)}
      inputProps={{ ...rest.inputProps, maxLength }}
      InputLabelProps={{
        ...rest.InputLabelProps,
        style: { fontSize: '12px' },
      }}
      FormHelperTextProps={{
        ...rest.FormHelperTextProps,
        style: { fontSize: '10px' },
      }}
    />
  </FieldWithError>
);

export default TextField;
