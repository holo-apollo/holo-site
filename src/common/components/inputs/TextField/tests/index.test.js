import React from 'react';
import renderer from 'react-test-renderer';

import TextField from '../index';

describe('TextField', function() {
  it('should match snapshot', function() {
    const tree = renderer
      .create(<TextField name="name" label="Label" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
