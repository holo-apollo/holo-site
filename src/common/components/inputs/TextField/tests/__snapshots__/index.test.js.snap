// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should match snapshot 1`] = `
<div
  className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
>
  <label
    className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
    data-shrink={false}
    style={
      Object {
        "fontSize": "12px",
      }
    }
  >
    Label
  </label>
  <div
    className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-46 MuiInput-input-31"
      disabled={false}
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;
